src/util/hash.cpp

```diff
@@ -67,4 +67,11 @@ unsigned hash_data(std::string const & data) {
     return hash(data.size(), [&] (unsigned i) { return static_cast<unsigned char>(data.data()[i]); });
 }
 
+unsigned combine_hashes(unsigned h1, unsigned h2) {
```

Comment:
Why not `hash(h1, h2)`?

Reply:
Do you know the rationale behind that algorithm? I used the one from SO because it had at least some explanation behind it, but on second thought `hash(h1, h2)` is probably fine, so I reverted to it.

Reply:
The rationale is very simple: Leo put it in, and so it's probably a good choice.  This is the commit where he added it in 2006: https://github.com/Z3Prover/z3/blob/e9eab22e5ccadc4c6ea36406e38d2dbc3d6786c2/lib/hash.h#L63-L68